File: frontend/src/types/UserChatStats.ts
/**
 * Response returned by the API for user chat statistics
 */
export interface UserChatStatsResponse {
  /**
   * Date of last message sent
   */
  lastChatDate: string;

  /**
   * Number of messages sent today
   */
  todayMessageCount: number;

  /**
   * Total number of messages sent all-time
   */
  totalMessageCount: number;

  /**
   * Maximum number of messages allowed per day for free users
   */
  maxFreeMessagesPerDay: number;

  /**
   * Whether the user has reached their daily message limit
   */
  hasReachedDailyLimit: boolean;
}

/**
 * Request to track a chat message
 */
export interface TrackMessageRequest {
  /**
   * Optional user ID. If not provided, the authenticated user's ID will be used.
   */
  userId?: string;

  /**
   * The session ID for the chat conversation.
   */
  sessionId?: string;
}

File: backend/Lithuaningo.API/Models/UserChatStats.cs
using Supabase.Postgrest.Attributes;
using Supabase.Postgrest.Models;

namespace Lithuaningo.API.Models
{
    /// <summary>
    /// Represents user chat usage statistics.
    /// </summary>
    [Table("user_chat_stats")]
    public class UserChatStats : BaseModel
    {
        /// <summary>
        /// Primary key.
        /// </summary>
        [PrimaryKey("id", false)]
        public Guid Id { get; set; }

        /// <summary>
        /// The user ID associated with these stats.
        /// </summary>
        [Column("user_id")]
        public Guid UserId { get; set; }

        /// <summary>
        /// Date of last chat message.
        /// </summary>
        [Column("last_chat_date")]
        public DateTime LastChatDate { get; set; }

        /// <summary>
        /// Number of messages sent today.
        /// </summary>
        [Column("today_message_count")]
        public int TodayMessageCount { get; set; }

        /// <summary>
        /// Total number of messages sent all-time.
        /// </summary>
        [Column("total_message_count")]
        public int TotalMessageCount { get; set; }
    }
}
